controllers/openstackcluster_controller.go

```diff
@@ -438,9 +438,109 @@ func (r *OpenStackClusterReconciler) reconcileNetworkComponents(log logr.Logger,
 }
 
 func (r *OpenStackClusterReconciler) SetupWithManager(mgr ctrl.Manager, options controller.Options) error {
-	return ctrl.NewControllerManagedBy(mgr).
+	controller, err := ctrl.NewControllerManagedBy(mgr).
 		WithOptions(options).
 		For(&infrav1.OpenStackCluster{}).
-		WithEventFilter(pausePredicates).
-		Complete(r)
+		WithEventFilter(pausedPredicates(r.Log)).
+		WithEventFilter(
+			predicate.Funcs{
+				// Avoid reconciling if the event triggering the reconciliation is related to incremental status updates
+				// for OpenStackCluster resources only
+				UpdateFunc: func(e event.UpdateEvent) bool {
+					if e.ObjectOld.GetObjectKind().GroupVersionKind().Kind != "OpenStackCluster" {
+						return true
+					}
+
+					oldCluster := e.ObjectOld.(*infrav1.OpenStackCluster).DeepCopy()
+					newCluster := e.ObjectNew.(*infrav1.OpenStackCluster).DeepCopy()
+
+					oldCluster.Status = infrav1.OpenStackClusterStatus{}
+					newCluster.Status = infrav1.OpenStackClusterStatus{}
+
+					oldCluster.ObjectMeta.ResourceVersion = ""
+					newCluster.ObjectMeta.ResourceVersion = ""
+
+					return !reflect.DeepEqual(oldCluster, newCluster)
+				},
+			},
+		).
+		Build(r)
+	if err != nil {
+		return errors.Wrap(err, "error creating controller")
+	}
+
+	return controller.Watch(
+		&source.Kind{Type: &clusterv1.Cluster{}},
+		&handler.EnqueueRequestsFromMapFunc{
+			ToRequests: handler.ToRequestsFunc(r.requeueOpenStackClusterForUnpausedCluster),
+		},
+		predicate.Funcs{
+			UpdateFunc: func(e event.UpdateEvent) bool {
+				oldCluster := e.ObjectOld.(*clusterv1.Cluster)
+				newCluster := e.ObjectNew.(*clusterv1.Cluster)
+				log := r.Log.WithValues("predicate", "updateEvent", "namespace", newCluster.Namespace, "cluster", newCluster.Name)
+				switch {
+				// return true if Cluster.Spec.Paused has changed from true to false
+				case oldCluster.Spec.Paused && !newCluster.Spec.Paused:
```

Comment:
I don't get why this code can be correct. Looks like we're now only reconciling unpause events? Any other update is just ignored? Do you know how this can be correct?

Reply:
Could it be that we avoid hitting this code via `requeueOpenStackClusterForUnpausedCluster`?

Reply:
Other update is still done by `WithEventFilter(pausedPredicates(r.Log))`  in L444.

Reply:
I am afraid I do not fully understand your second question, but Watch signature is the following so that this code is good to me.
```
Watch(src source.Source, eventhandler handler.EventHandler, predicates ...predicate.Predicate) error
```

Reply:
Ah so this means the watch in l.472 is an additional watch? So we reconcile both on the stuff which matches the filters in l.444 & l.445 and independently on the events that match our filteres in l.477?

I though all those filters are applied to the same event stream

Reply:
Yes, I understood as you said.

Reply:
Okay perfect :)